### ISSUE TITLE
False Positive: InlineDocCommentDeclarationSniff

### ISSUE THREAD

**Issue by JulianMar** [slevomat/coding-standard#1254]
I have a class like this, which defines callbacks which are executed in another context. For example, look at [laravels macro](https://levelup.gitconnected.com/understanding-laravel-macros-e2f493484a38) method. 

```
class Bar
{
	public function someFunction()
	{
		$callback1 = function () {
			/** @var Foo $this */
			$this->get();
		};

		$callback2 = function () {
			/** @var FooBar $this */
			$this->anything();
		};
	}
}
```

Now this gives me a false positive on the second $this assignment. 
`No assignment to $this variable before or after the documentation comment.` Which is weird, as it isn't an assignment. I pointed it down to the [isAssignment](https://github.com/slevomat/coding-standard/blob/28351a25af27f3a3e52c3adabefdddd7f0f63f95/SlevomatCodingStandard/Sniffs/Commenting/InlineD
ocCommentDeclarationSniff.php#L451) method. but I can't get behind the issue

I could do something like this but that's probably not a good solutions:
```
	private function isAssignment(File $phpcsFile, int $pointer): bool
	{
		$tokens = $phpcsFile->getTokens();

		$pointerAfterVariable = TokenHelper::findNextEffective($phpcsFile, $pointer + 1);
		if ($tokens[$pointerAfterVariable]['code'] === T_SEMICOLON ) {
			$pointerBeforeVariable = TokenHelper::findPreviousEffective($phpcsFile, $pointer - 1);
			return $tokens[$pointerBeforeVariable]['code'] === T_STATIC;
		}

		if ($tokens[$pointerAfterVariable]['code'] === T_OBJECT_OPERATOR) {
			return true;
		}

		return $tokens[$pointerAfterVariable]['code'] === T_EQUAL;
	}
```

Anyone has some advice? I can create a PR but I need some help there

**Comment by kukulich** [comment]
Improved in https://github.com/slevomat/coding-standard/commit/57c14d998bd4717f8267cc6096574084f2934f29
